client/state/jetpack/settings/utils.js

```diff
@@ -104,11 +102,17 @@ export const filterSettingsByActiveModules = ( settings ) => {
 	};
 	let filteredSettings = { ...settings };
 
-	forEach( moduleSettingsList, ( moduleSettings, moduleSlug ) => {
+	Object.entries( moduleSettingsList ).forEach( ( [ moduleSlug, moduleSettings ] ) => {
 		if ( ! settings[ moduleSlug ] ) {
-			filteredSettings = omit( filteredSettings, moduleSettings );
+			filteredSettings = Object.fromEntries(
+				Object.entries( filteredSettings ).filter(
+					( [ settingName ] ) => ! moduleSettings.includes( settingName )
+				)
+			);
 		}
-		filteredSettings = omit( filteredSettings, moduleSlug );
+		filteredSettings = Object.fromEntries(
+			Object.entries( filteredSettings ).filter( ( [ settingName ] ) => settingName !== moduleSlug )
+		);
 	} );
```

Comment:
This loop is so difficult to understand. Could it have some explanation comments? Or be somehow simplified? For example, it would be great to avoid deconstructing and constructing the `filteredSettings` object so many times, on every loop iteration.